Fix surface functions

matyas-streamhpc opened the pull request:
The changes on the generated document:
https://advanced-micro-devices-demo--3518.com.readthedocs.build/projects/HIP/en/3518/reference/kernel_language.html#surface-functions

lpaoletti reviewed (commented):
notice the file is named kernel_language.rst. 
that doesn't line up with the document title: C++ Language Extensions

Lining up the file name and title improves searchability.

matyas-streamhpc commented:
> notice the file is named kernel_language.rst. that doesn't line up with the document title: C++ Language Extensions
> 
> Lining up the file name and title improves searchability.

I am opening a new PR to fix it:
https://github.com/ROCm/HIP/pull/3534

neon60 committed:
Fix surface functions

Update surface functions in cpp language extensions
Update doxyfile and fix the function naming
Change to new clr branch
Fix white space characthers

neon60 commented:
Got approval. Merging to docs/develop.
